### ISSUE TITLE
False positives on class << self

### ISSUE THREAD

**Issue by dervus** [YorickPeterse/ruby-lint#145]
ruby: MRI 2.2.1
ruby-lint: 2.0.3

``` ruby
class FooBar
  class << self
    def foo
      5
    end

    def bar
      foo + foo
    end
  end
end
```
Output:
```
$ ruby-lint rubylint_bug.rb
rubylint_bug.rb: error: line 8, column 7: undefined method foo
rubylint_bug.rb: error: line 8, column 13: undefined method foo
```

**Comment by dervus** [comment]
Seems like it does not recognize `class << self` idiom at all, since if I substitute `foo` with `self.foo`, it prints:
```
$ ruby-lint rubylint_bug.rb
rubylint_bug.rb: error: line 8, column 7: undefined method foo on an instance of FooBar
rubylint_bug.rb: error: line 8, column 18: undefined method foo on an instance of FooBar
```
If I substitute `foo` with `self.class.foo`, it passes.

**Comment by mvidner** [comment]
I have added a (failing) test case for this in https://github.com/mvidner/ruby-lint/commit/8dcbc823795e91f6d5cb8fce21fb9a24a9787ba0 (branch https://github.com/mvidner/ruby-lint/tree/two-methods-in-sclass)
